lib/hydra/works/services/generic_file/full_text_extraction_service.rb

```diff
@@ -0,0 +1,33 @@
+module Hydra::Works
+  # Extract the full text from the content using Solr's extract handler
+  class FullTextExtractionService
+    def self.run(generic_file)
+      new(generic_file).extract
+    end
+
+    delegate :original_file, to: :@generic_file
+
+    def initialize(generic_file)
+      @generic_file = generic_file
+    end
+
+    def extract
+      uri = URI("#{connection_url}/update/extract?extractOnly=true&wt=json&extractFormat=text")
+      req = Net::HTTP.new(uri.host, uri.port)
+      resp = req.post(uri.to_s, original_file.content,           'Content-type' => "#{original_file.mime_type};charset=utf-8",
```

Comment:
Sometimes you have run rubocop multiple times as it will detect subsequent issues causes by its fixes to earlier ones.